src/sentry/auth/system.py

```diff
@@ -0,0 +1,110 @@
+from __future__ import absolute_import, print_function
+
+from django.conf import settings
+from django.utils.crypto import constant_time_compare
+from uuid import uuid4
+
+from sentry import options
+
+
+class SystemUser(object):
+    """
+    Singleton user representing a system that is part of Sentry.
+
+    The system user is a superuser that is not tied to an organization or real
+    user account. Unlike other superusers but similar to ``AnonymousUser``, it
+    has ``is_anonymous`` set.
+    """
+
+    id = -1  # Allow database queries
+
+    username = '<system>'
+    is_staff = True
+    is_active = True
+    is_superuser = True
+    is_system = True
```

Comment:
If the other user objects also implemented this you wouldn't need getattr() and we could more reliably ducktype users.

Reply:
I was trying to avoid exactly that, especially since we'd also have to patch this onto Django's own `AnonymousUser`. I'll leave leave this open for now and revisit it once the general approach has been agreed on.